aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -687,6 +692,250 @@ public void onError(Exception error) {
+                            ));
+                        }
+
+                        // Code verified user attribute was updated before
+                        if (result.size() == 0) {
+                            onSuccess.accept(new AuthUpdateAttributeResult(
+                                    true,
+                                    new AuthNextUpdateAttributeStep(
+                                            AuthUpdateAttributeStep.DONE,
+                                            Collections.emptyMap(),
+                                            null)
+                            ));
+                        }
+
+                        for (UserCodeDeliveryDetails details : result) {
+                            if (details.getAttributeName().equals(attribute.getKey().getKeyString())) {
+                                onSuccess.accept(new AuthUpdateAttributeResult(
+                                        true,
+                                        new AuthNextUpdateAttributeStep(
+                                                AuthUpdateAttributeStep.CONFIRM_ATTRIBUTE_WITH_CODE,
+                                                Collections.emptyMap(),
+                                                convertCodeDeliveryDetails(details))
+                                ));
+                            } else {
+                                onError.accept(new AuthException(
+                                        "Returned an incorrect attribute name",
+                                        "See attached exception for more details"
+                                ));
+                            }
+                        }
+                    }
+
+                    @Override
+                    public void onError(Exception error) {
+                        onError.accept(new AuthException(
+                                "Failed to update user attributes",
+                                error,
+                                "See attached exception for more details"
+                        ));
+                    }
+                }
+        );
+    }
+
+    @Override
+    public void updateUserAttributes(
+            @NonNull List<AuthUserAttribute> attributes,
+            @NonNull Consumer<Map<AuthUserAttributeKey, AuthUpdateAttributeResult>> onSuccess,
+            @NonNull Consumer<AuthException> onError
+    ) {
+        Map<String, String> attributesMap = new HashMap<>();
+        for (AuthUserAttribute attribute : attributes) {
+            attributesMap.put(attribute.getKey().getKeyString(), attribute.getValue());
+        }
+
+        awsMobileClient.updateUserAttributes(
+                attributesMap,
+                new Callback<List<UserCodeDeliveryDetails>>() {
+                    @Override
+                    public void onResult(List<UserCodeDeliveryDetails> result) {
+                        for (String attributeKey : attributesMap.keySet()) {
```

Comment:
Similar to above, the logic inside the onResult method should go something like this:

Create a resultMap<AuthUserAttributeKey, AuthUpdateAttributeResult> object upfront.

Loop through the objects inside the `result` object and add them to a codeDetailsMap<String, UserCodeDeliveryDetails> keyed by the attribute string name so you can easily check whether a given attribute exists in it. 

Loop through each attribute in the `attributes` list and for each one:

If it exists in the `codeDetailsMap` add an entry for it to `resultMap` with a status of CONFIRM_ATTRIBUTE_WITH_CODE
else add an entry to `resultMap` for it with a status of DONE.

There's no way there can be an error if we're inside the onResult callback.